Objects/object.c

```diff
@@ -2153,7 +2160,9 @@ new_reference(PyObject *op)
     if (_PyRuntime.tracemalloc.config.tracing) {
         _PyTraceMalloc_NewReference(op);
     }
-    Py_SET_REFCNT(op, 1);
+    /* Do not use Py_SET_REFCNT to skip the Immortal Instance check. This
+     * API guarantees that an instance will always be set to a refcnt of 1. */
+    op->ob_refcnt = 1;
```

Comment:
```suggestion
    /* Do not use Py_SET_REFCN, since it skips the Immortal Instance check.
     * This API guarantees that the refcnt will always be set to 1. */
    op->ob_refcnt = 1;
```

Reply:
Fixed!